### PR TITLE
Results for cluster list BDD

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Results for cluster list BDD

## Type of change

- New test feature file
- Non-breaking change in test steps implementation
- Refactor (refactoring code, removing useless files)
- Documentation update

## Testing steps

N/A

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary

### CODE REVIEW

File: docs/scenarios_list.md [hunk 1]
```diff
@@ -411,6 +411,12 @@ nav_order: 3
 * Check if list with two organizations is returned in case 'report' table contains reports for two organizations
 * Check if list with two organizations is returned in case 'report' table contains multiple reports for two organizations
 
+## `insights-results-aggregator/results_for_cluster_list.feature`
```
Comment (juandspy) [inline docs/scenarios_list.md:414]: I couldn't find this feature in https://github.com/RedHatInsights/insights-behavioral-spec/tree/main/features/insights-results-aggregator. is it missing?

File: docs/feature_list.md [hunk 2]
```diff
@@ -101,6 +101,7 @@ nav_order: 2
 * [Basic REST API endpoints provided by Insights Results Aggregator](https://github.com/RedHatInsights/insights-behavioral-spec/blob/main/features/insights-results-aggregator/basic_rest_api.feature)
 * [Checking responses from Insights Results Aggregator service: "organizations" endpoint](https://github.com/RedHatInsights/insights-behavioral-spec/blob/main/features/insights-results-aggregator/list_of_organizations.feature)
 * [Checking responses from Insights Results Aggregator service: "{organization}/clusters" endpoint](https://github.com/RedHatInsights/insights-behavioral-spec/blob/main/features/insights-results-aggregator/list_of_clusters.feature)
+* [Checking REST API endpoint that returns results for provided list of clusters](https://github.com/RedHatInsights/insights-behavioral-spec/blob/main/features/insights-results-aggregator/results_for_cluster_list.feature)
```
Comment (juandspy) [inline docs/feature_list.md:104]: couldn't find this link neither

Comment (tisnik) [reply docs/feature_list.md:104]: added

**Review by juandspy — APPROVED** [review]

**Review by joselsegura — APPROVED** [review]

File: features/steps/insights_results_aggregator_mock.py [hunk 3]
```diff
@@ -351,18 +351,29 @@ def check_metadata(context):
...
 
     # construct object to be send to the service
     json_request_body = {"clusters": cluster_list}
 
-    url = f"http://{context.hostname}:{context.port}{context.api_prefix}/clusters"
-
-    # perform POST request
-    context.response = requests.post(url, json=json_request_body)
+    # use RH identity token if needed
+    if org is not None and account is not None and user is not None:
```
Comment (dpensi) [inline features/steps/insights_results_aggregator_mock.py:364]: can be replace by `if org and account and user:`

**Review by dpensi — APPROVED** [review]

File: features/steps/insights_results_aggregator_mock.py (hunk 3 above)
Comment (dpensi) [reply features/steps/insights_results_aggregator_mock.py:364]: sorry, I didn't realized it was already merged! Nevermind!

**Pull request merged**